### PR TITLE
fix: Parse image tag for digest-based images

### LINKED ISSUE #51: Add support for digest image references
The input image reference currently requires a tag and does not accept a digest.

Example error with version 0.3.1 on linux:
```
dcp icr.io/cpopen/ibm-iam-operator-bundle@sha256:c63fdfbbeb3314d3ee889bee86139aa0bf6dd4a69f7e63139b6608abae0b3e5 -c manifests
 DEBUG dcp::image > 📦 Searching for image icr.io/cpopen/ibm-iam-operator-bundle@sha256:c63fdfbbeb3314d3ee889bee86139aa0bf6dd4a69f7e63139b6608abae0b3e5 locally
Error: ❌ error pulling image: error 400 Bad Request - invalid reference format
```

### PR DESCRIPTION
**Pull request by exdx**
Supports digest-based images

Closes #51

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -72,3 +77,23 @@ pub async fn is_present_locally_podman(images: PodmanImages, search_for_image: S
     }
     false
 }
+
+#[cfg(test)]
```
Comment (tylerslaton) [inline src/image.rs:81]: Should this live in the `tests` directory?

**Review by tylerslaton — COMMENTED** [review]
Talked about this a bit offline, but perhaps we can get the detection of locally pulled images working for digests?

File: src/image.rs (hunk 1 above)
Comment (exdx) [reply src/image.rs:81]: I think its idiomatic to have unit tests to live alongside the code its testing in Rust

Comment (tylerslaton) [reply src/image.rs:81]: Ah right, those are E2E tests. Good call.

**Comment by exdx** [comment]
> Talked about this a bit offline, but perhaps we can get the detection of locally pulled images working for digests?

I think I might tackle this in a follow-up and keep this PR small in scope.

**Pull request merged**